crates/polars-arrow/src/io/ipc/read/stream.rs

```diff
@@ -172,12 +162,18 @@ fn read_next<R: Read>(
                 None,
                 dictionaries,
                 metadata.version,
-                &mut reader,
+                &mut reader.take(block_length as u64),
                 0,
-                file_size,
                 scratch,
             );
 
+            let new_pos = reader.stream_position()?;
```

Comment:
Can you make the reborrow explicit here? That is, pass `(&mut reader).take(block_length as u64)` instead? Right now this only works because `reader.take` automatically gets transformed to `(&mut *reader).take`.

Reply:
I have change this to `&mut (&mut *reader).take(block_length as u64)`. But I still think it's a little verbose...